__tests__/07_nested_map_spec.ts

```diff
@@ -112,4 +112,39 @@ describe('issue #14', () => {
     await Promise.resolve();
     expect(map1.get('items').get('item1').get('color')).toStrictEqual('red');
   });
+
+  it('triggers a limited set of updates on bind', async () => {
+    const proxy1 = proxy<Record<string, any>>({ items: {} });
+    const proxy2 = proxy<Record<string, any>>({ items: {} });
+
+    const doc1 = new Y.Doc();
+    const doc2 = new Y.Doc();
+
+    const map1 = doc1.getMap('map') as any;
+    const map2 = doc2.getMap('map') as any;
+
+    const listener1 = jest.fn();
+    const listener2 = jest.fn();
+
+    doc1.on('update', (update: Uint8Array) => {
+      Y.applyUpdate(doc2, update);
+      listener1();
+    });
+    doc2.on('update', (update: Uint8Array) => {
+      Y.applyUpdate(doc1, update);
+      listener2();
+    });
+
+    bind(proxy1, map1);
+    proxy1.items.item1 = { color: 'red' };
+    proxy1.items.item2 = { color: 'red' };
+    proxy1.items.item3 = { color: 'red' };
+
+    await Promise.resolve();
+    bind(proxy2, map2);
+    await Promise.resolve();
+
+    expect(listener1).toBeCalledTimes(2);
```

Comment:
This was the original issue I was running into. Update would be triggered n+1 times depending on the number of items. I believe there was a missing equality check. Binding to a relatively large initial ydoc was giving me: `Caught error while handling a Yjs update RangeError: Maximum call stack size exceeded`